xbmc/filesystem/File.cpp

```diff
@@ -815,9 +815,9 @@ ssize_t CFile::Write(const void* lpBuf, size_t uiBufSize)
     { // "test" write with zero size
       // some VFSs don't handle correctly null buffer pointer
       // provide valid buffer pointer for them
-      auto_buffer dummyBuf(255);
-      dummyBuf.get()[0] = 0;
-      return m_pFile->Write(dummyBuf.get(), 0);
+      std::vector<char> dummyBuf(255);
+      dummyBuf.data()[0] = 0;
+      return m_pFile->Write(dummyBuf.data(), 0);
```

Comment:
I would change this to
```
const char dummyData = 0;
return m_pFile->Write(&dummyData, 0);
```
No need to construct a `std::vector` and to allocate on the heap.

Reply:
updated